src/main/java/se/sundsvall/byggrintegrator/service/ByggrIntegratorService.java

```diff
@@ -106,25 +109,43 @@ public Weight getErrandType(String caseNumber) {
 	}
 
 	@Cacheable("listNeighborhoodNotificationFilesCache")
-	public String listNeighborhoodNotificationFiles(String municipalityId, String identifier, String caseNumber) {
+	public String listNeighborhoodNotificationFiles(final String municipalityId, final String identifier, final String caseNumber, final String referralReference) {
 		// Fetch errand from ByggR
 		final var result = byggrIntegration.getErrand(caseNumber);
 
-		// Prefix identifier if it contains organisation legal id and add hyphen to identifier as ByggR integration formats
+		// Extract referralReferenceId
+		final var referralReferenceId = Optional.ofNullable(referralReference)
```

Comment:
I don't know if it's possible that a "]"-char can occur before any "["-char, but if so, there might be a problem...
Maybe use pattern matching instead?

Something like:
```
final var pattern = Pattern.compile("\\[(\\d+)]");
final int referralReferenceId = Optional.ofNullable(referralReference)
        .map(pattern::matcher)
        .filter(Matcher::find)
        .map(m -> Integer.parseInt(m.group(1)))
        .orElse(0);
```

Reply:
Its not possible.